flake.nix

```diff
@@ -16,8 +16,15 @@
         overlays = [ rust-overlay.overlays.default ];
         pkgs = import nixpkgs { inherit system overlays; };
         rust = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+        isDarwin = system == "x86_64-darwin" || system == "aarch64-darwin";
+         # Add firefox deps only on non-darwin.
+         # darwin is listed in badPlatforms in pkgs.firefox's meta.
+        optionalPackages = if isDarwin then [] else [
+          pkgs.firefox
+          pkgs.geckodriver
+        ];
```

Comment:
I've typically seen things done like this before, would it work here? 

```
          buildInputs = [
            pkgs.bitcoind
            clightning-dev
            pkgs.openssl
            pkgs.zlib
            pkgs.postgresql
            pkgs.gcc
            pkgs.pkg-config
            pkgs.poetry
            pythonEnv
            pkgs.gcc.cc.lib
          ] ++ pkgs.lib.optionals pkgs.stdenv.isDarwin [
            pkgs.libiconv
          ];
        };
```

Reply:
It works. pushed the change